src/tests/Model/ResultSummaryCreatorTests.cs

```diff
@@ -0,0 +1,162 @@
+﻿using NUnit.Framework;
+using System;
+
+namespace NUnit.Gui.Model
+{
+    [TestFixture]
+    public class ResultSummaryCreatorTests
+    {
+        [Test]
+        public void WhenResultNodeIsNotForTestRunExceptionIsThrown()
+        {
+            Assert.That(() => CreateResultSummary("<anything-other-than-test-run/>"), Throws.InstanceOf<InvalidOperationException>());
+
+            Assert.That(() => CreateResultSummary("<test-run/>"), Throws.Nothing);
+        }
+        
+        [Test]
+        public void WhenResultIsNotSpecified_PassedIsDefault()
+        {
+            var summary = CreateResultSummary("<test-run/>");
+
+            Assert.That(summary.OverallResult, Is.EqualTo("Passed"));
+
+            summary = CreateResultSummary("<test-run result='Failed'/>");
+
+            Assert.That(summary.OverallResult, Is.EqualTo("Failed"));
+        }
+
+        [Test]
+        public void WhenDurationIsNotSpecified_ZeroIsDefault()
+        {
+            var summary = CreateResultSummary("<test-run/>");
+
+            Assert.That(summary.Duration, Is.EqualTo(0.0));
+
+            summary = CreateResultSummary("<test-run duration='1.9'/>");
+
+            Assert.That(summary.Duration, Is.EqualTo(1.9));
+        }
+
+        [Test]
+        public void WhenStartTimeIsNotSpecified_DateTimeMinValueIsDefault()
+        {
+            var summary = CreateResultSummary("<test-run/>");
+
+            Assert.That(summary.StartTime, Is.EqualTo(DateTime.MinValue));
+
+            var expectedDate = new DateTime(2017, 7, 8, 6, 19, 23);
+            summary = CreateResultSummary($"<test-run start-time='{expectedDate.ToString("u")}'/>");
+
+            Assert.That(summary.StartTime, Is.EqualTo(expectedDate));
+        }
+
+        [Test]
+        public void WhenEndTimeIsNotSpecified_DateTimeMaxValueIsDefault()
+        {
```

Comment:
Perhaps write this as two separate tests? Then we can also use a TestCase for EndTime to test other endtimes.